x/group/spec/messages.md

```diff
@@ -0,0 +1,46 @@
+```go
+// Groups get their own GroupID
+type MsgCreateGroup struct {
+	Signer sdk.AccAddress `json:"signer"`
+	// The Owner of the group is allowed to change the group structure. A group account
+	// can own a group in order for the group to be able to manage its own members
+	Owner  sdk.AccAddress `json:"owner"`
+	// The members of the group and their associated weight
+	Members []Member `json:"members,omitempty"`
+	// TODO maybe make this something more specific to a domain name or a claim on identity? or Info leave it generic
+	Memo string `json:"memo,omitempty"`
+}
+
+// group accounts get their own sdk.AccAddress
+type MsgCreateGroupAccount struct {
+	Signer         sdk.AccAddress `json:"signer"`
+	// The Owner of a group account is allowed to change the DecisionPolicy. This can be left nil 
+	// in order for the group account to "own" itself
+	Owner          sdk.AccAddress `json:"owner"`
+	Group          GroupID        `json:"group"`
+	DecisionPolicy DecisionPolicy `json:"decision_policy"`
+	Memo           string         `json:"memo,omitempty"`
+}
+type MsgCreateProposal struct {
+	Proposer sdk.AccAddress `json:"proposer"`
+	GroupAccount    sdk.AccAddress `json:"group"`
+	Msgs     []sdk.Msg      `json:"msgs"`
+	// Exec can be set to true in order to attempt to execute the proposal immediately
```

Comment:
Alex: this is the way to unify single-tx multisig, with on-chain governance rules, rather than have two different modules. 

Since you worked on those both separately in weave, please reflect if there are any issues that arise by this unifying abstraction

Reply:
I do understand the motivation but there will be quite some complexity for any UI client that have to show the state or votes. I don't know much about the veto rules, yet but they may add complexity, too.